src/main/scala/com/gu/googleauth/auth.scala

```diff
@@ -43,7 +44,9 @@ case class GoogleAuthConfig private(
   enforceValidity: Boolean,
   prompt: Option[String],
   antiForgeryChecker: AntiForgeryChecker
-)
+) {
+  val impliesEndUserHasHTTPSConnection: Boolean = new URI(redirectUrl).getScheme == "https"
```

Comment:
OAuth 2.x requires TLS (it doesn't encrypt the params) so this is technically redundant.

Reply:
> OAuth 2.x requires TLS (it doesn't encrypt the params) so this is technically redundant.

In Dev, on the Ophan team at least, we aren't using HTTPS - and the sign-in process with Google OAuth _works_, even if it isn't secure. So our `redirectUrl` is configured to `http://localhost:9000/oauth2callback` - just plain HTTP - and this means that `impliesEndUserHasHTTPSConnection` would be `false` - and so we wouldn't want to add HSTS headers.

Reply:
IIRC if you have a HSTS header on a non-HTTPS website then it will get ignored anyway.

Reply:
> IIRC if you have a HSTS header on a non-HTTPS website then it will get ignored anyway.

Ah, you're right, it does - thanks!

> **Note:** The `Strict-Transport-Security` header is **ignored** by the browser when your site is accessed using HTTP; this is because an attacker may intercept HTTP connections and inject the header or remove it.  When your site is accessed over HTTPS with no certificate errors, the browser knows your site is HTTPS capable and will honor the `Strict-Transport-Security` header.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security#Description

Reply:
Yep, as @sihil says the HSTS header is not the same as redirecting to secure. With that in mind maybe you see why I'm uncomfortable with the header magically being added by a dependency? It's just a weird thing to do.

Did you mean for this change to be doing the secure redirect in PROD as well?

Reply:
> Did you mean for this change to be doing the secure redirect in PROD as well?

No, I was happy enough to get the browser to do the redirect rather than having to open up port 80 on the ELB!